sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/translation/SqlOperators.java

```diff
@@ -91,6 +92,12 @@
           x -> createTypeFactory().createArrayType(x.getOperandType(0), -1),
           new UdafImpl<>(new ArrayAgg.ArrayAggArray()));
 
+  public static final SqlOperator ARRAY_CONCAT_AGG_FN =
+      createUdafOperator(
+          "array_concat_agg",
+          x -> createTypeFactory().createArrayType(x.getOperandType(0), -1),
```

Comment:
I'm pretty sure the problem is with this line here. The return type is supposed to be ARRAY<BIGINT>, but the type returned here is ARRAY<ARRAY<BIGINT>>. I'm guessing it's because `x.getOperandType(0)` is an array type. Instead it should be the type of elements inside `x.getOperandType(0)`. (I'm not sure exactly what the method is to get that, but you should be able to figure it out using the debugger.)

Reply:
@ibzib  changing` ARRAY<ARRAY<BIGINT>>` to` ARRAY<BIGINT>` from works, thanks for your help
```
  x ->  createTypeFactory()
                  .createArrayType(createTypeFactory().createSqlType(SqlTypeName.BIGINT), -1),
```